force_torque_sensor_controller/src/force_torque_sensor_controller.cpp

```diff
@@ -0,0 +1,171 @@
+    return controller_interface::return_type::ERROR;
+  }
+
+  return controller_interface::return_type::SUCCESS;
+}
+
+CallbackReturn ForceTorqueSensorController::on_configure(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  state_interface_names_ = node_->get_parameter("state_interface_names").as_string_array();
+  if (state_interface_names_.empty()) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'state_interface_names' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  sensor_name_ = node_->get_parameter("sensor_name").as_string();
+  if (sensor_name_.empty()) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'sensor_name' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  frame_id_ = node_->get_parameter("frame_id").as_string();
+  if (frame_id_.empty()) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'frame_id' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  fx_range_ = node_->get_parameter("fx_range").as_int();
+  if (fx_range_ == -1) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'fx_range' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  tz_range_ = node_->get_parameter("tz_range").as_int();
+  if (tz_range_ == -1) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'tz_range' parameter was empty");
+    return CallbackReturn::ERROR;
+  }
+
+  RCLCPP_INFO_STREAM(get_node()->get_logger(), "configure successful");
+
+  try {
+    // register ft sensor data publisher
+    sensor_state_publisher_ = get_node()->create_publisher<geometry_msgs::msg::WrenchStamped>(
+        "force_torque_sensor", rclcpp::SystemDefaultsQoS());
+  } catch (...) {
+    return CallbackReturn::ERROR;
+  }
+
+  return CallbackReturn::SUCCESS;
+}
+
+controller_interface::InterfaceConfiguration
+ForceTorqueSensorController::command_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration command_interfaces_config;
+  command_interfaces_config.type = controller_interface::interface_configuration_type::NONE;
+
+  return command_interfaces_config;
+}
+
+controller_interface::InterfaceConfiguration
+ForceTorqueSensorController::state_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration state_interfaces_config;
+  state_interfaces_config.type = controller_interface::interface_configuration_type::INDIVIDUAL;
+
+  for (const auto & interface : state_interface_names_) {
+    state_interfaces_config.names.push_back(sensor_name_ + "/" + interface);
+  }
+
+  return state_interfaces_config;
+}
+
+CallbackReturn ForceTorqueSensorController::on_activate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  return CallbackReturn::SUCCESS;
+}
+
+CallbackReturn ForceTorqueSensorController::on_deactivate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  return CallbackReturn::SUCCESS;
+}
+
+controller_interface::return_type ForceTorqueSensorController::update()
+{
+  geometry_msgs::msg::Vector3 f_vec;
```

Comment:
Why not work directly with the wrench message?

Reply:
I made some modifications. Please check once and suggest if this is ok.